### PR TITLE
 For your review: Update index.js

### PR DESCRIPTION
**Pull request by akileez**
change variable "content" to "substance" avoiding conflict with {{content}} block helper.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -54,14 +54,14 @@ module.exports.register = function(Handlebars, options, params) {
...
+        context: processContext(grunt, _.defaults({substance: content}, context)),
+        substance: content
       };
     }).sort(compare_fn).map(function (obj) {
 
       // promote id into context
       obj.context.id = obj.id;
-      var template = Handlebars.compile(options.fn(obj.context) + obj.content);
+      var template = Handlebars.compile(options.fn(obj.context));
       return template(obj.context);
```
Comment (jonschlinkert) [inline index.js:65]: Should we be returning a `Handlebars.SafeString` here?

Comment (akileez) [inline index.js:65]: Maybe... I need to test globbing files. I was thinking about my use case of single file glob. 
A few lines below is the `Handlebars.SafeString` call. Whether the SafeString needs to be within the `html` block or not I will test now. 

``` js
      // promote id into context
      obj.context.id = obj.id;
      var template = Handlebars.compile(options.fn(obj.context));
      return template(obj.context);

    }).join(options.sep);

    return new Handlebars.SafeString(html);
  });
```

So

Comment (akileez) [inline index.js:65]: Tested with file glob (approx 25 files) and everything works including using separator. 

In the last update, you refactored the block in question from a `return glob.find(cwd(src), options...` statement where the `Handlebars.SafeString` resided within to a `var html = grunt.file.expand(options.glob, cwd(src))...` statement where the SafeString is outside of this block.

The changes presented are not affected by the refactor and all appears to be working exactly how I expect. The only question I have is: Do you want to use the word `substance` for `content`? (Or any other word for that matter)
I used `zulu` when testing.

**Comment by jonschlinkert** [comment]
Lol, `zulu`. I do the same kind of thing pretty often... 

When a helper and variable have the same name, Handlebars always gives precedence to the helper. However, you can easily get around by using `this` to tell Handelbars what the context should be for the variable, e.g. `{{this.content}}`, so technically that's all that needs to happen here. 

I'm open to whatever, we can use any variable that makes sense there, but perhaps the best option is  to use `Handlebars.createFrame` so we can add `content` to the `data` object of the helper, which would enable `{{@content}}` to be used in the templates. IMO, this is a more idiomatic solution but I might not be able to do it for a couple days... so if you need something to be merged in to solve an immediate need then I'll be practical about it and just get it done

**Comment by akileez** [comment]
You do whatever is best and take as much time as possible to fix properly. My brain is settled now. Thank you for the info with `{{this.content}}`. Every little bit helps. I will await your changes to learn better methods of coding  Handlebars/javascript. Sounds interesting. You can withdraw/close this request.

**Comment by jonschlinkert** [comment]
You're PR was really good, please feel 100% comfortable submitting pull requests here any time. This dialog helps me as well, everything is a learning experience.

**Comment by jonschlinkert** [comment]
btw, here is an example of `createFrame` being used. 
- https://github.com/helpers/handlebars-helper-eachitems/blob/master/eachItems.js#L37
- https://github.com/helpers/handlebars-helper-eachitems/blob/master/eachItems.js#L44-L49

Any variables on the `data` object can now be referenced in templates with: 

```
{{@index}}
{{@number}}
{{@first}}
{{@prev}}
{{@next}}
```

**Comment by akileez** [comment]
kewl. thanks. but it is gonna take me a little while to wrap my brain around these concepts. more foreign stuff for my grey matter to digest.

**Comment by jonschlinkert** [comment]
closing. pushing up some fixes

**Pull request closed without merging**